storage_policies/README.md

```diff
@@ -0,0 +1,266 @@
+# Storage policies
+
+This document describes where our production datasets are stored, how [object
+lifecycles] are configured, and how access permissions are managed.
+
+[object lifecycles]: https://cloud.google.com/storage/docs/lifecycle
+
+We are trying to strike a balance between:
+
+- Quick development iterations and unlimited ad-hoc data exploration.
+- Robust, reproducible pipelines, using only strictly necessary cloud
+  resources.
+
+This motivates two somewhat unusual principles in the design:
+
+- Quick development iterations and testing only happens on a small (but
+  representative) subset of the data (blue highlight in the graph below). The
+  full dataset is only accessible through code that has been reviewed and
+  committed (green highlight in the graph below).
+- All outputs are versioned and immutable, except for purely temporary
+  results (yellow highlight in the graph below). Since "production runs" of
+  pipelines only happen after sufficient testing on subsets of the data,
+  immutable results generally shouldn't cause a lot of churn or resource
+  usage.
+
+## Typical data flow
+
+![data flow](images/dataflow.svg)
+
+## Buckets
+
+In this context, a stack corresponds to a particular project / effort, e.g.
+*TOB-WGS* or *RDNow*, with separate buckets and permission groups. Below,
+`$STACK` is a placeholder for the name of that effort, e.g. `$STACK` =
+`tob-wgs`.
+
+Currently, all buckets reside in the `australia-southeast1` GCP region. It's
+therefore essential that all computation happens in that region too, to avoid
+network egress costs.
+
+In general, all datasets within buckets should be versioned, using a simple
+major-minor naming scheme like `gs://cpg-$STACK-main/qc/v1.2/`. We don't have
```

Comment:
We started using https://github.com/iterative/dvc at UMCCR for data versioning, which may or may not interest you. Essentially it's like git for datasets (though I'll admit I haven't played with it too much).
Say you have v1.2 of a dir containing 10 files. And say in dir v1.3 you only update 1 of those files (e.g. add 5 rows), but the other 9 files are same between v1.2 and v1.3 i.e. you're duplicating data. Something like DVC would just keep track of the changed files (so would keep the 9 files, and the 2 versions of the single changed file).

Reply:
Yes, DVC looks really cool and I've discussed this a bit with the Hail team :).

Unfortunately in our case with data in GCP buckets (way too large to store in git directly), it doesn't quite work (as far as I can tell).